src/components/FormDialog/index.js

```diff
@@ -16,11 +16,52 @@ import useStyles from "./useStyles";
 
 import CtAButton from "@/promisetracker/components/CtAButton";
 
-function FormDialog({ open, handleFormClose, ...props }) {
+function FormDialog({ session, open, handleFormClose, ...props }) {
   const classes = useStyles(props);
-
   const { petitionTitle, petitionDescription } = props;
+  const [values, setValues] = React.useState({
+    title: "",
+    description: "",
+    recipients: "",
+    problemStatement: "",
+    numberOfSignaturesRequired: null,
+    image: "",
+    video: "",
+    source: "",
+  });
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    values.source = { link: values.source };
+    values.number_of_signatures_required = Number(
+      values.numberOfSignaturesRequired
+    );
+    values.problem_statement = values.problemStatement;
+    delete values.problemStatement;
+    delete values.numberOfSignaturesRequired;
+
+    // temporary
+    delete values.image;
+    delete values.video;
 
+    fetch("/api/petitions", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${session.accessToken}`,
+      },
+      body: JSON.stringify(values),
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw new Error("Network response was not ok.");
+      })
```

Comment:
the lib function to create does something similar (i.e passing bearer using session from api request)